common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -0,0 +1,37 @@
+package whisk.connector.kafka
+
+import akka.actor.ActorSystem
+import scaldi.Injector
+import whisk.common.Logging
+import whisk.core.WhiskConfig
+import whisk.core.connector.MessageConsumer
+import whisk.core.connector.MessageProducer
+import whisk.core.connector.MessagingProvider
+import whisk.spi.SpiFactoryModule
+/**
+  * Created by tnorris on 6/20/17.
+  */
+class KafkaMessagingProvider(actorSystem:ActorSystem, config:WhiskConfig)(implicit logging:Logging) extends MessagingProvider {
```

Comment:
What I still don't get on OpenWhisk.. is why not call this MessagingPRovider, and Kafka being the implementation?

If you want to make it pluggable, make it pluggable...